### PR TITLE
Updates vmci_client C library

### PR DESCRIPTION
**Pull request by venilnoronha**
* Extracted header file from vmci_client.c
* Updated vmci_client impl to support Winsock
* Added vmci_client.def for DLL export

### CODE REVIEW

File: esx_service/vmci/vmci_client.h [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+//
+
+typedef struct be_request {
+   uint32_t mlen;   // length of message (including trailing \0)
+   const char *msg; // null-terminated immutable JSON string.
+} be_request;
+
+#define MAXBUF 1024 * 1024 // Safety limit. We do not expect json string > 1M
+#define MAX_CLIENT_PORT 1023 // Last privileged port
+#define START_CLIENT_PORT 100 // Where to start client port
```
Comment (shahzebnp) [inline esx_service/vmci/vmci_client.h]: minor: Since you are moving this content, lets have #defines first and then all struct definitions.

**Review by shahzebnp — APPROVED** [review]

File: esx_service/vmci/vmci_client.h (hunk 1 above)
Comment (venilnoronha) [reply esx_service/vmci/vmci_client.h]: sounds good

File: esx_service/vmci/vmci_client.h [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+//
+typedef struct {
+   int sock_id; // socket id for socket APIs
+   struct sockaddr_vm addr; // held here for bookkeeping and reporting
+} be_sock_id;
+
+//
+// Protocol message structure: request and reply
+//
+
```
Comment (shaominchen) [inline esx_service/vmci/vmci_client.h]: Nit: remove the empty line.

File: esx_service/vmci/vmci_client.h [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+#include "vmci_sockets.h"
+#include "connection_types.h"
+
+#define ERR_BUF_LEN 512
+#define MAXBUF 1024 * 1024 // Safety limit. We do not expect json string > 1M
+#define MAX_CLIENT_PORT 1023 // Last privileged port
+#define START_CLIENT_PORT 100 // Where to start client port
+#define BIND_RETRY_COUNT (MAX_CLIENT_PORT - START_CLIENT_PORT) // Retry entire range on bind failures
+
+// operations status. 0 is OK
```
Comment (shaominchen) [inline esx_service/vmci/vmci_client.h]: Nit: operations... => Operations..

File: esx_service/vmci/vmci_client.c [hunk 4]
```diff
@@ -18,52 +18,16 @@
 //
 // Called mainly from Go code.
 //
-// API: Exposes only Vmci_GetReply. The call is blocking.
+// API: Exposes Vmci_GetReply and Vmci_FreeBuf. The calls are blocking.
 //
+// For details on _WIN32 specific handling, please refer the following link:
```
Comment (shaominchen) [inline esx_service/vmci/vmci_client.c]: Nit: refer => refer to

**Review by shaominchen — APPROVED** [review]
The refactoring change looks good. A few trivial nits in existing code.

**Pull request merged**